packages/framework/agent-scheduler/src/scheduler.ts

```diff
@@ -420,8 +422,11 @@ export class AgentSchedulerFactory implements IFluidDataStoreFactory {
 
     public static async createChildInstance(parentContext: IFluidDataStoreContext): Promise<AgentScheduler> {
         const packagePath = [...parentContext.packagePath, AgentSchedulerFactory.type];
-        const router = await parentContext.containerRuntime.createDataStore(packagePath);
-        return requestFluidObject<AgentScheduler>(router, "/");
+        const dataStore = await parentContext.containerRuntime.createDataStore(packagePath);
+        const maybeHandle: FluidObject<IFluidHandle> = (dataStore as any);
+        const handle = await maybeHandle.IFluidHandle?.get();
+        assert(handle instanceof AgentScheduler, "The data store's handle is not an AgentScheduler!");
```

Comment:
instanceof may backfire (things like passing objects cross-iframe objects).
If you could avoid it by doing some other check, that would be advised.